### DISCUSSION TITLE
reuse social link icon in custom component

### DISCUSSION THREAD

**Discussion by Sepush** [vuejs/vitepress#1123] [Q&A]
Is it possible to reuse the social link icon of vitepres in custom components?
like `import {Github} from 'vitepress/icon'`
Is there already an API?
I found there is `icons` and `support/icones` in `vitepress client` how about expose them

**Comment by brc-dd** [comment]
We have picked [these](https://github.com/vuejs/vitepress/blob/main/src/client/theme-default/support/socialIcons.ts) from https://github.com/simple-icons/simple-icons, you can directly install that.

**Reply by Sepush** [reply]
great how about expose more component like `VPSocialLink` `VPSocialLinks`

**Reply by brc-dd** [reply]
@Sepush 

```ts
import VPSocialLinks from 'vitepress/client/theme-default/components/VPSocialLinks.vue'
```

**Reply by Sepush** [reply]
thanks I will try it

**Reply by Sepush** [reply]
it works but with a type error
![image](https://user-images.githubusercontent.com/39197136/183112494-baa5d4fc-1030-4ba9-869d-f53bb217c45f.png)

**Reply by brc-dd** [reply]
Probably our typings are messed up (created an issue to track it -- #1127). For now, you can consider adding this to your `tsconfig.json`:

```ts
{
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "vitepress/client/*": ["node_modules/vitepress/dist/client/*"]
    }
  },
  "include": ["**/*", "**/*.vue"]
}
```

**Reply by Sepush** [reply]
thanks

**Reply by Sepush** [reply]
this seems broken in alpha8
![image](https://user-images.githubusercontent.com/39197136/185725783-3796db0e-1674-4cc6-be98-64d233977399.png)

**Reply by brc-dd** [reply]
Use:

```js
import VPSocialLinks from 'vitepress/dist/client/theme-default/components/VPSocialLinks.vue'
// or
import VPSocialLinks from '@theme/components/VPSocialLinks.vue'
```

Second one will work only when you don't have custom theme (also in TS it would need some path alias).

Just a word of caution, this component is internal. We'll likely be changing the way it works in future releases. Currently the bundle includes unused svg codes, we'll probably be transforming the icon name to svg during build for performance reasons. So this component in future will need the svg code. I recommend writing a custom component (and registering it globally for ease).

**Reply by Sepush** [reply]
got it
